### ISSUE TITLE
[Feature]: Add logits for classification task

### ISSUE THREAD

**Issue by orionw** [vllm-project/vllm#12910]
### 🚀 The feature, motivation and pitch

I noticed that for models with only single class output of true/false (e.g. [RankLLaMA](https://huggingface.co/castorini/rankllama-v1-7b-lora-passagehttps://huggingface.co/castorini/rankllama-v1-7b-lora-passage)) because it only one class output everything from vLLM is marked as true, even though the logit may be below zero.

I don't see how to change how this functions in the code, so it would be great to get the logits so that this could be done by the user. 

### Alternatives

You could provide a threshold value or something for classification for single-class prediction outputs as well.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
As mentioned [here](https://docs.vllm.ai/en/latest/models/pooling_models.html#pooling-models), you can disable softmax via `--override-pooling-config` option to get the original logits.

**Comment by orionw** [comment]
Awesome, didn't realize that pooling was used for classification. That should work great then!

**Comment by orionw** [comment]
For reference @DarkLight1337 do you know where we can do this programmatically to remove it? Is it in the LLM class, just setting `override-pooling-config=None`? I couldn't find any docs on it.

**Comment by DarkLight1337** [comment]
For offline inference, you can set `override_pooling_config=PoolingConfig(...)`
